custom_metrics/privacy.js

```diff
@@ -1,24 +1,312 @@
 //[privacy]
 // Uncomment the previous line for testing on webpagetest.org
 
+// README! Instructions for adding a new custom metric for the Web Almanac.
+// 2. If the value requires more than one line of code, evaluate it in an IIFE, eg `(() => { ... })()`. See `link-nodes`.
+// 3. Test your change by following the instructions at https://github.com/HTTPArchive/almanac.httparchive.org/issues/33#issuecomment-502288773.
+// 4. Submit a PR to update this file.
+
+const response_bodies = $WPT_BODIES;
+
+/**
+ * @function testPropertyStringInResponseBodies
+ * Test that a JS property string is accessed in response bodies
+ * (given that wrapping properties to log accesses is not possible as metrics run at the end)
+ * only in Document and Script resources (HTML/JS)
+ * inspired by https://github.com/HTTPArchive/legacy.httparchive.org/blob/master/custom_metrics/event-names.js
+ *
+ * @param {string} pattern - Regex pattern to match in the response bodies.
+ * @return {boolean} - True, if pattern was matched.
+ */
+function testPropertyStringInResponseBodies(pattern) {
+  try {
+    let re = new RegExp(pattern);
+    return response_bodies
+      .filter(body => body.type === 'Document' || body.type === 'Script')
```

Comment:
I'd suggest moving this filtering out of the function so that it's only done once as opposed to each time the function is called. (response bodies are very large so these optimizations can be significant, especially after running them on 8M+ pages)